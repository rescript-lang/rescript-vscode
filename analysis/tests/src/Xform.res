type kind = First | Second | Third | Fourth(int)
type r = {name: string, age: int}

let ret = _ => assert(false)
let kind = assert(false)

if kind == First {
  // ^xfm
  ret("First")
} else {
  ret("Not First")
}

#kind("First", {name: "abc", age: 3}) != kind ? ret("Not First") : ret("First")
//             ^xfm

let name = "hello"
//   ^xfm

let annotated: int = 34
//   ^xfm

module T = {
  type r = {a: int, x: string}
}

let foo = x =>
  //      ^xfm
  switch x {
  | None => 33
  | Some(q) => q.T.a + 1
  //     ^xfm
  }

let withAs = (~x as name) => name + 1
//                   ^xfm

@react.component
let make = (~name) => React.string(name)
//   ^xfm

let _ = (~x) => x + 1
//       ^xfm

//
// Add braces to the body of a function
//

let noBraces = () => name
//                   ^xfm

let nested = () => {
  let _noBraces = (_x, _y, _z) => "someNewFunc"
  //                              ^xfm
}

let bar = () => {
  module Inner = {
    let foo = (_x, y, _z) =>
      switch y {
      | #some => 3
      | #stuff => 4
      }
    //^xfm
  }
  Inner.foo(1, ...)
}

module ExtractableModule = {
  /** Doc comment. */
  type t = int
  // A comment here
  let doStuff = a => a + 1
  // ^xfm
}

let variant = First

let _x = switch variant {
| First => "first"
| _ => "other"
//  ^xfm
}

let _x = switch variant {
| First | Second => "first"
| _ => "other"
//  ^xfm
}

let _x = switch variant {
| First if 1 > 2 => "first"
| Second => "second"
| _ => "other"
//  ^xfm
}

let polyvariant: [#first | #second | #"illegal identifier" | #third(int)] = #first

let _y = switch polyvariant {
| #first => "first"
| _ => "other"
//  ^xfm
}

let _y = switch polyvariant {
| #first | #second => "first"
| _ => "other"
//  ^xfm
}

let variantOpt = Some(variant)

let _x = switch variantOpt {
| Some(First) => "first"
| _ => "other"
//  ^xfm
}

let _x = switch variantOpt {
| Some(First) | Some(Second) => "first"
| _ => "other"
//  ^xfm
}

let _x = switch variantOpt {
| Some(First | Second) => "first"
| _ => "other"
//  ^xfm
}

let polyvariantOpt = Some(polyvariant)

let _x = switch polyvariantOpt {
| Some(#first) => "first"
| None => "nothing"
| _ => "other"
//  ^xfm
}

let _x = switch polyvariantOpt {
| Some(#first | #second) => "first"
| _ => "other"
//  ^xfm
}
