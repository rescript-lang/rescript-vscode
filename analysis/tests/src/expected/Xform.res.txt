Xform src/Xform.res 6:5
posCursor:[6:3] posNoWhite:[6:1] Found expr:[6:0->11:1]
Completable: Cpath Value[kind]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[kind]
Path kind
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Replace with switch

TextDocumentEdit: Xform.res
{"start": {"line": 6, "character": 0}, "end": {"line": 11, "character": 1}}
newText:
<--here
switch kind {
| First =>
  // ^xfm
  ret("First")
| _ => ret("Not First")
}

Xform src/Xform.res 13:15
Hit: Replace with switch

TextDocumentEdit: Xform.res
{"start": {"line": 13, "character": 0}, "end": {"line": 13, "character": 79}}
newText:
<--here
switch kind {
| #kind("First", {name: "abc", age: 3}) => ret("First")
| _ => ret("Not First")
}

Xform src/Xform.res 16:5
Hit: Add type annotation

TextDocumentEdit: Xform.res
{"start": {"line": 16, "character": 8}, "end": {"line": 16, "character": 8}}
newText:
        <--here
        : string
Hit: Add Documentation template

TextDocumentEdit: Xform.res
{"start": {"line": 16, "character": 0}, "end": {"line": 16, "character": 18}}
newText:
<--here
/**

*/
let name = "hello"

Xform src/Xform.res 19:5
Hit: Add Documentation template

TextDocumentEdit: Xform.res
{"start": {"line": 19, "character": 0}, "end": {"line": 19, "character": 23}}
newText:
<--here
/**

*/
let annotated: int = 34

Xform src/Xform.res 26:10
Hit: Add type annotation

TextDocumentEdit: Xform.res
{"start": {"line": 26, "character": 10}, "end": {"line": 26, "character": 11}}
newText:
          <--here
          (x: option<T.r>)

Xform src/Xform.res 30:9
Hit: Add braces to function

TextDocumentEdit: Xform.res
{"start": {"line": 26, "character": 0}, "end": {"line": 32, "character": 3}}
newText:
<--here
let foo = x => {
  //      ^xfm
  switch x {
  | None => 33
  | Some(q) => q.T.a + 1
  //     ^xfm
  }
}

Xform src/Xform.res 34:21
Hit: Add type annotation

TextDocumentEdit: Xform.res
{"start": {"line": 34, "character": 24}, "end": {"line": 34, "character": 24}}
newText:
                        <--here
                        : int

Xform src/Xform.res 38:5
Hit: Add Documentation template

TextDocumentEdit: Xform.res
{"start": {"line": 37, "character": 0}, "end": {"line": 38, "character": 40}}
newText:
<--here
/**

*/
@react.component
let make = (~name) => React.string(name)

Xform src/Xform.res 41:9
Hit: Add type annotation

TextDocumentEdit: Xform.res
{"start": {"line": 41, "character": 11}, "end": {"line": 41, "character": 11}}
newText:
           <--here
           : int

Xform src/Xform.res 48:21
posCursor:[48:21] posNoWhite:[48:19] Found expr:[48:15->48:25]
posCursor:[48:21] posNoWhite:[48:19] Found expr:[48:15->48:25]
Completable: Cpath Value[name]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[name]
Path name
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Add braces to function

TextDocumentEdit: Xform.res
{"start": {"line": 48, "character": 0}, "end": {"line": 48, "character": 25}}
newText:
<--here
let noBraces = () => {
  name
}

Xform src/Xform.res 52:34
Hit: Add braces to function

TextDocumentEdit: Xform.res
{"start": {"line": 51, "character": 0}, "end": {"line": 54, "character": 1}}
newText:
<--here
let nested = () => {
  let _noBraces = (_x, _y, _z) => {
    "someNewFunc"
  }
  //                              ^xfm
}

Xform src/Xform.res 62:6
Hit: Add braces to function

TextDocumentEdit: Xform.res
{"start": {"line": 58, "character": 4}, "end": {"line": 62, "character": 7}}
newText:
    <--here
    let foo = (_x, y, _z) => {
      switch y {
      | #some => 3
      | #stuff => 4
      }
    }

Xform src/Xform.res 72:5
Hit: Extract module as file

CreateFile: ExtractableModule.res

TextDocumentEdit: ExtractableModule.res
{"start": {"line": 0, "character": 0}, "end": {"line": 0, "character": 0}}
newText:
<--here
/** Doc comment. */
type t = int
// A comment here
let doStuff = a => a + 1
// ^xfm


TextDocumentEdit: src/Xform.res
{"start": {"line": 68, "character": 0}, "end": {"line": 74, "character": 1}}
newText:
<--here


Xform src/Xform.res 80:4
posCursor:[78:16] posNoWhite:[78:14] Found expr:[78:9->82:1]
Completable: Cpath Value[variant]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[variant]
Path variant
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Expand catch-all

TextDocumentEdit: Xform.res
{"start": {"line": 80, "character": 2}, "end": {"line": 80, "character": 3}}
newText:
  <--here
  Second | Third | Fourth(_)

Xform src/Xform.res 88:4
posCursor:[86:16] posNoWhite:[86:14] Found expr:[86:9->90:1]
Completable: Cpath Value[polyvariant]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[polyvariant]
Path polyvariant
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Expand catch-all

TextDocumentEdit: Xform.res
{"start": {"line": 88, "character": 2}, "end": {"line": 88, "character": 3}}
newText:
  <--here
  #second | #"illegal identifier" | #third(_)

Xform src/Xform.res 96:4
posCursor:[94:16] posNoWhite:[94:14] Found expr:[94:9->98:1]
Completable: Cpath Value[variantOpt]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[variantOpt]
Path variantOpt
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Expand catch-all

TextDocumentEdit: Xform.res
{"start": {"line": 96, "character": 2}, "end": {"line": 96, "character": 3}}
newText:
  <--here
  Some(Second | Third | Fourth(_)) | None

Xform src/Xform.res 105:4
posCursor:[102:16] posNoWhite:[102:14] Found expr:[102:9->107:1]
Completable: Cpath Value[polyvariantOpt]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[polyvariantOpt]
Path polyvariantOpt
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
Hit: Expand catch-all

TextDocumentEdit: Xform.res
{"start": {"line": 105, "character": 2}, "end": {"line": 105, "character": 3}}
newText:
  <--here
  Some(#"illegal identifier" | #second | #third(_))

